File: CS/CustomChooseReportTypePage.cs
using System;
using System.Collections.Generic;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard;
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.Entity.ProjectModel;
using DevExpress.XtraReports.Wizards;
using DevExpress.XtraReports.Wizards.Presenters;
using DevExpress.XtraReports.Wizards.Views;

namespace CustomReportWizard {
    class CustomChooseReportTypePage : ChooseReportTypePage<XtraReportModel> {
        public CustomChooseReportTypePage(IChooseReportTypePageView view, IConnectionStorageService connectionStorageService,
            DataSourceTypes dataSourceTypes, IWizardRunnerContext context, ISolutionTypesProvider solutionTypesProvider) : 
            base(view, connectionStorageService, dataSourceTypes, context, solutionTypesProvider) { }

        public override bool MoveNextEnabled {
            get { return true; }
        }

        public override Type GetNextPageType() {
            if(View.ReportType == ReportType.Standard || View.ReportType == ReportType.Empty)
                return typeof(ChoosePageSettingsPage<CustomReportModel>);
            return base.GetNextPageType();
        }
    }
}

File: VB/CustomChooseReportTypePage.vb
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Wizard
Imports DevExpress.DataAccess.Wizard.Services
Imports DevExpress.Entity.ProjectModel
Imports DevExpress.XtraReports.Wizards
Imports DevExpress.XtraReports.Wizards.Presenters
Imports DevExpress.XtraReports.Wizards.Views
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CustomReportWizard
    Friend Class CustomChooseReportTypePage
        Inherits ChooseReportTypePage(Of XtraReportModel)

        Public Sub New(ByVal view As IChooseReportTypePageView, ByVal connectionStorageService As IConnectionStorageService, ByVal dataSourceTypes As DataSourceTypes, ByVal context As IWizardRunnerContext, ByVal solutionTypesProvider As ISolutionTypesProvider)
            MyBase.New(view, connectionStorageService, dataSourceTypes, context, solutionTypesProvider)
        End Sub

        Public Overrides ReadOnly Property MoveNextEnabled() As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides Function GetNextPageType() As Type
            If View.ReportType = ReportType.Standard OrElse View.ReportType = ReportType.Empty Then
                Return GetType(ChoosePageSettingsPage(Of CustomReportModel))
            End If
            Return MyBase.GetNextPageType()
        End Function
    End Class
End Namespace
